### DISCUSSION TITLE
TBeam V1.1 Sx1276 Flashing issue

### DISCUSSION THREAD

**Discussion by tbeam915** [markqvist/Reticulum#276] [Help & Questions]
Hello,

When I run rnodeconf -a on my LilyGO T-Beam I get the following output 

`
Is the above correct? [y/N] y
[15:39:00] Checking firmware file availability...
[15:39:01] The latest firmware for this board is version 1.59
[15:39:01] Using existing firmware file: rnode_firmware_tbeam.zip for version 1.59
[15:39:01] Verifying firmware integrity...
[15:39:01] Decompressing firmware...
Archive:  /home/joek/.config/rnodeconf/update/1.59/rnode_firmware_tbeam.zip
  inflating: /home/joek/.config/rnodeconf/update/1.59/esptool.py  
  inflating: /home/joek/.config/rnodeconf/update/1.59/console_image.bin  
  inflating: /home/joek/.config/rnodeconf/update/1.59/rnode_firmware_tbeam.boot_app0  
  inflating: /home/joek/.config/rnodeconf/update/1.59/rnode_firmware_tbeam.bin  
  inflating: /home/joek/.config/rnodeconf/update/1.59/rnode_firmware_tbeam.bootloader  
  inflating: /home/joek/.config/rnodeconf/update/1.59/rnode_firmware_tbeam.partitions  
[15:39:01] Firmware decompressed
[15:39:01] Flashing RNode firmware to device on /dev/ttyACM0
esptool.py v3.1
Serial port /dev/ttyACM0
Connecting......
Chip is ESP32-D0WDQ6-V3 (revision 3)
Features: WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Crystal is 40MHz
MAC: 08:3a:f2:67:9c:34
Uploading stub...
Running stub...
Stub running...
Changing baud rate to 921600
Changed.
Configuring flash size...
Flash will be erased from 0x0000e000 to 0x0000ffff...
Flash will be erased from 0x00001000 to 0x00005fff...
Flash will be erased from 0x00010000 to 0x0018cfff...
Flash will be erased from 0x00210000 to 0x003fffff...
Flash will be erased from 0x00008000 to 0x00008fff...
Compressed 8192 bytes to 47...
Wrote 8192 bytes (47 compressed) at 0x0000e000 in 0.1 seconds (effective 612.0 kbit/s)...
Hash of data verified.
Compressed 17472 bytes to 12125...
Wrote 17472 bytes (12125 compressed) at 0x00001000 in 0.5 seconds (effective 302.7 kbit/s)...
Hash of data verified.
Compressed 1558432 bytes to 1011775...
Wrote 1558432 bytes (1011775 compressed) at 0x00010000 in 14.2 seconds (effective 878.5 kbit/s)...
Hash of data verified.
Compressed 2031616 bytes to 1828548...
Wrote 2031616 bytes (1828548 compressed) at 0x00210000 in 22.5 seconds (effective 722.8 kbit/s)...

A fatal error occurred: MD5Sum command returned unexpected result: b'\x91\x18\x99\xa4\xd0\x0e=\xacm2\rr\xff\xfe\xc5'
`
I found the following discussion https://github.com/markqvist/Reticulum/discussions/259

which details out how to pull older firmware down and create a hash file to attempt to install it. I have done that with both 1.57 and 1.58 and obtain the same fatal error.

Not sure if I can pull older versions of the rnodeconf through pip but that would be the next work around.

**Comment by htd-02** [comment]
Hey !
Thanks for mentioning your discussion in my issue thread, looks like your issue may be the same as me and a few others, you can try the fix I explained in [270](https://github.com/markqvist/Reticulum/issues/270)

But make sure to change the baud rate in the right place, line 2007 is for the specific board I was using, should be line 1935 for the T-Beam. The baud rates that were recommended were 460800, 230400 and 115200 instead of the default 921600.

I hope this works for you !

**Comment by markqvist** [comment]
It may be the case that this occurs for boards with specific types of USB-to-serial converter chips, that can't handle the higher baudrates. Do you know what serial converter your board uses?

**Comment by cobraPA** [comment]
I just saw similar with a T-Beam V1.1 I picked up.  I dropped the relevant baud setting to 115200 to test and it's writing.  I'm on Mac and saw:
Configuring flash size...

A fatal error occurred: Timed out waiting for packet content
Flashing succeeded at the lower baud rate.  Serial chip is claimed to be CH9102 on the web page for the board, but I don't see that chip on the surface of the T-Beam.

**Reply by markqvist** [reply]
I've seen this a couple of times now on Mac, where some chips will not work above 115200. It's an issue with the drivers included in macOS. Downloading an updated driver package from the Chinese manufacturers website alleviates the problem, but may feel a bit daunting to some ;)

**Reply by cobraPA** [reply]
Oh, wow, add the chinese drivers... interesting fix if a little scary.  Thanks for the heads up...

**Reply by markqvist** [reply]
Hahaha, yeah, that's the thing ;)

I'll add a slow-boat mode to `rnodeconf` at some point, so there's just a command-line switch to drop it into 115200 baud mode, but have not gotten around to it yet.

**Comment by pizdjuk** [comment]
Same here with LilyGo v2.1. But this did not occure before. I tried to flash RNode original FW, and it failed. Then I tried to flash the version for lilygo v2.1 and it failed as described, but worked anyway. Only rnode claims that the signature is not the correct one.

**Reply by cobraPA** [reply]
Was your host machine also Mac?

**Reply by pizdjuk** [reply]
what is Mac? Apple? No, I have Intel AMD64 with Linux

**Reply by markqvist** [reply]
If you want help, you need to specify exactly *what* is failing. Just saying "same here, it failed" is too little information for anyone to be able to help you.

What error message do you get? Where do you get it? The terminal output? On the device itself? At what point in the process? What choices did you make in the autoinstaller menu? What *exact* board model do you have?
